src/decorators/safe-url.decorator.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidationArguments,
+} from 'class-validator';
+import { Transform } from 'class-transformer';
+import * as sanitizeHtml from 'sanitize-html';
```

Comment:
**praise**: Good choice of using `sanitize-html` to ensure the URL is free from malicious content.